### PR TITLE
Check for proper namespace/var definitions

### PR DESCRIPTION
**Pull request by porkostomus**
This PR adds basic functions for determining if the namespace and main function have been defined* correctly. 

I'm going about this by trying to think of anything that could be wrong with a solution, starting with the obvious. Does it sound like I'm on the right track?

"*" i.e. *named* correctly, referring to the main function. We're not even talking about the *implementation* yet.

### CODE REVIEW

File: src/exercism/clojure_analyzer.clj [hunk 1]
```diff
@@ -7,6 +7,28 @@
...
+        (map :name (:namespace-definitions (ana file)))))
+
+(defn var-defined? [s file]
+  (some #(= (symbol s) %)
+        (map :name (:var-definitions (ana file)))))
+ 
+(comment
+  (ns-defined? "two-fer" "resources/example.clj")
+  (var-defined? "two-fer" "resources/example.clj")
+  )
```
Comment (ErikSchierboom) [inline src/exercism/clojure_analyzer.clj:28]: What does this function call do?

**Review by ErikSchierboom — APPROVED** [review]
> I'm going about this by trying to think of anything that could be wrong with a solution, starting with the obvious. Does it sound like I'm on the right track?

Yep, that's totally fine. Another way to think about analyzers is to have them detect the most common mistakes students make first. Getting automatic feedback on commonly made mistakes greatly helps reduce the burden on mentors. As such, it's much better to have analyzers that detect the most common mistakes for the 10 most submitted exercises, than it is to detect _all_ mistakes in a couple of exercises.

File: src/exercism/clojure_analyzer.clj (hunk 1 above)
Comment (porkostomus) [reply src/exercism/clojure_analyzer.clj:28]: That's what is called a "Rich comment form". It's used for evaluating functions during interactive development, but the whole block is ignored by the compiler when the file is read.

Comment (porkostomus) [reply src/exercism/clojure_analyzer.clj:28]: more info: https://github.com/PEZ/rich4clojure

Comment (ErikSchierboom) [reply src/exercism/clojure_analyzer.clj:28]: That's neat!

Comment (porkostomus) [reply src/exercism/clojure_analyzer.clj:28]: I don't know how much Clojure you've done so I'm sorry if you know this already, but all development is done with an editor connected to the running program, which is built by evaluating functions (often one at a time) using editor hotkeys. So Clojure files are more than source code, they're like playgrounds.

Comment (ErikSchierboom) [reply src/exercism/clojure_analyzer.clj:28]: I've been doing Clojure in IntelliJ with the Cursive plugin, so I think that my workflow has been a bit different.

**Pull request merged**